pkg/kubelet/util/pluginwatcher/example_handler.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluginwatcher
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+
+	v1beta1 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta1"
+	v1beta2 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta2"
+)
+
+var chanForHandlerAckTesting chan error = make(chan error)
+var callbackCount = struct {
+	mutex sync.Mutex
+	count int32
+}{}
+
+func exampleHandler(pluginName string, endpoint string, versions []string, sockPath string) (error, chan bool) {
```

Comment:
totally agree, i am trying to avoid test code in example_handler.go, i implement a logic of checking duplicate plugin name as a example, this is not for testing, because the actual handler will likely need to handle this.

also i still keep the supported versions as hard coded, the actual handler may do similar thing or make it configurable via different methods.

hope this make sense to you, thanks